lib/Models/CompositeCatalogItem.js

```diff
@@ -25,6 +35,20 @@ var CompositeCatalogItem = function(terria, items) {
 
 	this.items = defined(items) ? items.slice() : [];
 
+    overrideProperty(this, 'legendUrls', {
```

Comment:
In order for Knockout dependency tracking to work properly, `knockout.defineProperty` (which `overrideProperty` calls) must come _after_ `knockout.track`.   See the "Computed Properties" section here:
http://blog.stevensanderson.com/2013/05/20/knockout-es5-a-plugin-to-simplify-your-syntax/